src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+import React, { Component } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { HtmlDiv } from '../../../../reset';
+import { windowAvailable } from '../../../../utils/common';
+import { AutoId } from '../../../../utils/AutoId';
+import { Debounce } from '../../../utils/Debounce/Debounce';
+import { Button } from '../../../Button/Button';
+import { Chip } from '../../../Chip';
+import { Popover } from '../../../Popover/Popover';
+import { FilterInput, FilterInputStatus } from '../../FilterInput/FilterInput';
+import { ComboboxItemList } from '../ComboboxItemList/ComboboxItemList';
+import { ComboboxItem } from '../ComboboxItem/ComboboxItem';
+import { ComboboxEmptyItem } from '../ComboboxEmptyItem/ComboboxEmptyItem';
+import { ChipList } from '../ChipList/ChipList';
+import { baseStyles } from './Combobox.baseStyles';
+
+const baseClassName = 'fi-combobox';
+const comboboxClassNames = {
+  wrapper: `${baseClassName}_wrapper`,
+  open: `${baseClassName}--open`,
+  removeAllButton: `${baseClassName}_removeAllButton`,
+  error: `${baseClassName}--error`,
+};
+
+export interface ComboboxData {
+  /** Unique label that will be shown on combobox item and used on filter */
+  labelText: string;
+  /** Using labelText if chipText is not given */
+  chipText?: string;
+  /** Item selection disabled for the user */
+  disabled?: boolean;
+  /** Unique id to identify the item */
+  uniqueItemId: string;
+}
+
+export interface ComboboxProps<T extends ComboboxData> {
+  /** Combobox container div class name for custom styling. */
+  className?: string;
+  /** Items for the combobox */
+  items: Array<T & ComboboxData>;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Label */
+  labelText: string;
+  /** Event that is sent when item selections change */
+  onItemSelectionsChange?: (selectedItems: Array<T>) => void;
+  /** Show chip list */
+  chipListVisible?: boolean;
```

Comment:
have we used this form of naming with booleans? I'm thinking `showChipList` instead, but not sure what would be in line with other components.

Reply:
Will need to be checked against other components using booleans.

Somehow `showChipList`seems wrong to be for prop name as it starts with an action and therefore associated to be function (in my mind at least :D )

Reply:
True, naming semantics are difficult. On the other hand, pretty often action start with something referring to an event, onChange, onBlur, onSelectItem and so on. Found a couple of references that proposed using is/has/can prefix for booleans. The most important thing in my opinion is to be consistent within this library - same conventions should be used everywhere.

Reply:
As discussed elsewhere, this is probably mostly in line with what we have in other components. So I propose we leave it as is.